cmake/jemalloc.cmake

```diff
@@ -20,11 +20,21 @@ MACRO (CHECK_JEMALLOC)
       SET(CMAKE_REQUIRED_LIBRARIES pthread dl m)
       SET(what bundled)
     ELSE()
-      SET(libname jemalloc)
+      IF(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" AND
+        CMAKE_SYSTEM_VERSION STRGREATER "10.0")
+        SET(libname c)
+      ELSE()
+        SET(libname jemalloc)
+      ENDIF()
       SET(what system)
     ENDIF()
 
-    CHECK_LIBRARY_EXISTS(${libname} malloc_stats_print "" HAVE_JEMALLOC)
+    IF(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" AND
+      CMAKE_SYSTEM_VERSION STRGREATER "10.0")
+      SET(HAVE_JEMALLOC ON)
+    ELSE()
+      CHECK_LIBRARY_EXISTS(${libname} malloc_stats_print "" HAVE_JEMALLOC)
+    ENDIF()
```

Comment:
On FreeBSD (as of 10.0) jemalloc is in the base libc so checking malloc_stats_print in a library fails. Earlier patches checked it with libname libc iirc.